sagemaker_batch_transform/tensorflow_open-images_tfrecord/tensorflow-serving-tfrecord-python-sdk.ipynb

```diff
@@ -0,0 +1,385 @@
+    "\n",
+    "The code below creates a SageMaker Model entity that will be used for Batch inference, and runs a Transform Job using that Model. The Model contains a reference to the TFS container, and the `model.tar.gz` containing our TensorFlow SavedModel and the pre- and post-processing `inference.py` script."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import sagemaker\n",
+    "from sagemaker.tensorflow.serving import Model\n",
+    "\n",
+    "s3_path = 's3://{}/{}'.format(bucket, prefix)\n",
+    "\n",
+    "model_data = sagemaker_session.upload_data('model.tar.gz',\n",
+    "                                           bucket,\n",
+    "                                           os.path.join(prefix, 'model', 'tfrecord'))\n",
+    "                                           \n",
+    "tensorflow_serving_model = Model(model_data=model_data,\n",
+    "                                 role=role,\n",
+    "                                 framework_version='1.13',\n",
+    "                                 sagemaker_session=sagemaker_session)\n",
+    "\n",
+    "input_path = 's3://sagemaker-sample-data-{}/batch-transform/open-images/tfrecord'.format(region)\n",
+    "\n",
+    "print('Model data S3 path: {}'.format(model_data))\n",
+    "print('Input S3 path: {}'.format(input_path))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Before we create a Transform Job, let's inspect some of our input data. Here's an example, the first image in our dataset. The data in the input path consists of 100 TFRecord files, each with 1,000 JPEG images of varying sizes and shapes. Here is a subset:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!echo \"Transform input path: {input_path}\"\n",
+    "!aws s3 ls {input_path}/ --human-readable"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can inspect the format of each TFRecord file. The first record in the object named \"train-00001-of-00100\" refers to object \"785877fb88018e89.jpg\":\n",
+    "\n",
+    "<img src=\"sample_image/785877fb88018e89.jpg\">"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!aws s3 cp s3://sagemaker-sample-data-{region}/batch-transform/open-images/tfrecord/train-00001-of-00100 .\n",
+    "import tensorflow as tf\n",
+    "iterator = tf.python_io.tf_record_iterator(\"train-00001-of-00100\")\n",
+    "example = next(iterator)\n",
+    "result = tf.train.Example.FromString(example)\n",
+    "message_map = result.ListFields()[0][1]\n",
+    "\n",
+    "print(message_map)"
```

Comment:
This will print the full b64 encoded byte stream - Did you mean to include this?

Reply:
Yes -- this is meant to show an example input, and the output cell is truncated so it won't fill up the notebook.